### PR TITLE
Try dark toolbar for the write mode

### LINKED ISSUE #66054: Use the previous "select mode" dark toolbar style for Write mode
Let's try adding an additional affordance to write mode to help you understand what "mode" you're in, writing or design, by leveraging the pre-existing styles for select mode. This way there's a clear visual affordance to indicate the difference in experiences. 

### Visual 

Image

Related #65298.

### PR DESCRIPTION
**Pull request by dhruvang21**
## What?

Solves #66054

Let's try adding an additional affordance to write mode to help you understand what "mode" you're in, writing or design, by leveraging the pre-existing styles for select mode. This way there's a clear visual affordance to indicate the difference in experiences.

## Why?
Part of [66054](https://github.com/WordPress/gutenberg/issues/66054)

## How?
I’ve applied the conditions for the `is-contentonly-mode `class, which gets added when the write mode is activated. I also tried to add the SCSS to match the design. While I did my best to ensure the SCSS is correct, I believe it can be further optimized through the review process if necessary.

## Testing Instructions
 1. Open a post or page.
 2. Insert a any block or pattern.
 3. Then change the mode to write mode see the toolbar in darkmode.

## Screenshots or screencast 

https://github.com/user-attachments/assets/b7e6e5d5-15ab-4e85-8712-52ed656ecee9

### CODE REVIEW

**Comment by dhruvang21** [comment]
@richtabor can you please review this PR

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 1]
```diff
@@ -57,6 +57,25 @@
...
+			color: $white;
+		}
+		.components-toolbar-group,
+		.components-toolbar {
+			border-right-color: $gray-600 !important;
+		}
+		.is-pressed {
+			color: var(--wp-admin-theme-color);
+		}
+	}
```
Comment (draganescu) [inline packages/block-editor/src/components/block-toolbar/style.scss]: We used to have a dark toolbar which was removed here:

https://github.com/WordPress/gutenberg/pull/65204/files#diff-70b1ee574a01ebb2f11bc23c4aa937951549171851aa03e9c930310f20762de6L91

Can we reuse the styles there? It may tbe that is a better foundation to start with?

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 2]
```diff
@@ -57,6 +57,25 @@
 		}
 	}
 
+	&.is-contentonly-mode {
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-toolbar/style.scss]: Can we use is-dark-toolbar class to be more explicit. There might be different modes that would want to trigger that behavior.

Comment (dhruvang21) [reply packages/block-editor/src/components/block-toolbar/style.scss]: @youknowriad updated the classname in the latest commit.

File: packages/block-editor/src/components/block-toolbar/style.scss (hunk 1 above)
Comment (dhruvang21) [reply packages/block-editor/src/components/block-toolbar/style.scss]: @draganescu  added the these styles into block-tools style.scss file please review it again.

File: packages/block-editor/src/components/block-tools/style.scss [hunk 3]
```diff
@@ -118,15 +118,30 @@
 		}
 	}
 
-	.block-editor-block-toolbar {
```
Comment (dhruvang21) [inline packages/block-editor/src/components/block-tools/style.scss]: I accidentally removed this CSS in the previous commit. I’ll add it back in the next commit along with any other changes (if needed).

File: packages/block-editor/src/components/block-toolbar/style.scss [hunk 4]
```diff
@@ -93,6 +93,10 @@
 		box-shadow: 0 $border-width 0 0 rgba($color: #000, $alpha: 0.133); // 0.133 = $gray-200 but with alpha.
 	}
 
+	&.is-contentonly-mode {
+		background-color: $gray-900;
+	}
```
Comment (youknowriad) [inline packages/block-editor/src/components/block-toolbar/style.scss]: Is this change related, I'm guessing it's a lingering change right?

**Comment by youknowriad** [comment]
This is working decently well for me. Would love a design review to ensure it's inline what the original ideal.

File: packages/block-editor/src/components/block-toolbar/index.js [hunk 5]
```diff
@@ -167,11 +171,13 @@ export function PrivateBlockToolbar( {
 	// Shifts the toolbar to make room for the parent block selector.
 	const classes = clsx( 'block-editor-block-contextual-toolbar', {
 		'has-parent': showParentSelector,
+		'is-dark-toolbar': ! isDefaultEditingMode && ! hasFixedToolbar,
```
Comment (jameskoster) [inline packages/block-editor/src/components/block-toolbar/index.js]: I'd avoid words that dictate the appearance (`dark`) as we may need to re-evaluate the design when we introduce light/dark modes down the road.

**Comment by jameskoster** [comment]
I haven't been very close to this work so take my feedback with a grain of salt, but I don't know that I'd equate a different toolbar color to a different _mode_. I'd be tempted to try something with the canvas... maybe in design mode you see the artboard and resize handles:

Screenshot 2024-10-24 at 14 01 59

---

In terms of the execution I don't love the style overrides; they generally lead to maintenance headaches down the road.

Toggled buttons are looking a bit odd and don't meet contrast requirements:

Screenshot 2024-10-24 at 13 51 38

Menus and popovers remain light, is this intentional?

Screenshot 2024-10-24 at 13 52 06

---

In an ideal world this would be handled more sustainably, IE at the component level via theming.

File: packages/block-editor/src/components/block-toolbar/index.js (hunk 5 above)
Comment (youknowriad) [reply packages/block-editor/src/components/block-toolbar/index.js]: So how you would name this. See my previous comment here https://github.com/WordPress/gutenberg/pull/66116#discussion_r1810685860

**Comment by jasmussen** [comment]
I would echo Jay's note. There are some good ideas, good intents. But compent and admin-level theming is increasingly on the radar, which will afford a systematic approach to more easily test and tune this in a way that does not risk adding technical debt. I would circle back to this once we're a bit further. But thank you for the contribution, the instinct, and the work!

**Comment by jasmussen** [comment]
My above comment aside, 66054 makes a good case for why a dark toolbar as outlined can make sense even before theming. If this passes a good code review, no blockers from my end.

File: packages/block-editor/src/components/block-toolbar/style.scss (hunk 4 above)
Comment (dhruvang21) [reply packages/block-editor/src/components/block-toolbar/style.scss]: yes i have updated the code.

**Comment by jameskoster** [comment]
I certainly agree it's worth trying _something_ to help indicate the active mode. I'm not 100% sold on using the toolbar to do that because it's not always visible, and because 'top toolbar' exists. That said I would also not block :)

File: packages/block-editor/src/components/block-toolbar/index.js (hunk 5 above)
Comment (getdave) [reply packages/block-editor/src/components/block-toolbar/index.js]: We could use `inverted` or `alt` (as in `alternative`)?

**Comment by richtabor** [comment]
Pushed a few slight color changes. 

CleanShot 2024-10-25 at 11 53 02

**Review by getdave — COMMENTED** [review]
Thanks for this PR.

Screenshot 2024-10-25 at 10 56 39

For me when the toolbar is over a black background it provides a poor experience. I suspect this might be a problem for a11y as well.

I think we _need_ to have some kind of outline on the toolbar to ensure it stands out on darker background.

**Comment by jasmussen** [comment]
Related, though not for this PR: #66455.

File: packages/block-editor/src/components/block-toolbar/index.js [hunk 6]
```diff
@@ -137,6 +140,7 @@ export function PrivateBlockToolbar( {
 			isUsingBindings: _isUsingBindings,
 			hasParentPattern: _hasParentPattern,
 			hasContentOnlyLocking: _hasTemplateLock,
+			hasFixedToolbar: getSettings().hasFixedToolbar,
```
Comment (richtabor) [inline packages/block-editor/src/components/block-toolbar/index.js:143]: Why was this added?

**Comment by richtabor** [comment]
> I think we need to have some kind of outline on the toolbar to ensure it stands out on darker background.

Good call, updated: 

CleanShot 2024-10-25 at 15 11 31

**Comment by jameskoster** [comment]
We also need to fix the contrast on toggled buttons.

**Comment by jameskoster** [comment]
Spotted another issue. Select a template part in the page editor and hover the name; it disappears:

Screenshot 2024-10-31 at 09 34 47

**Review by draganescu — COMMENTED** [review]
There is a bug in the implementation which keeps the light toolbar around after switching modes, plus the style leaks to _all nstances of content only_ whereas we only want this for the **write mode**


https://github.com/user-attachments/assets/f5df1c6a-97a5-47aa-889d-2cd03e27a164

**Comment by dhruvang21** [comment]
Hello @draganescu,

Could you please share the steps to recreate the issue you showed in the video? I wasn’t able to reproduce it when testing in the Gutenberg PR reviewer.

File: packages/block-editor/src/components/block-toolbar/index.js (hunk 5 above)
Comment (draganescu) [reply packages/block-editor/src/components/block-toolbar/index.js]: @dhruvang21 I think this line causes the issues in my video - `isDefaultEditingMode` applies to indivudual blocks which can be locked or in partially synced patterns so we can end up with the dark toolbar in default mode.

Comment (dhruvang21) [reply packages/block-editor/src/components/block-toolbar/index.js]: @draganescu, you’re absolutely right; this condition isn’t complete as it fails for blocks with unique states—like those that are locked, reusable, or partially synced in patterns—where each block’s mode can differ within the editor.

For now, I think we could add this condition: `getBlockEditingMode(parentClientId) === 'contentOnly'`, which directly targets the parent block’s mode. However, a better approach might be to retrieve the editor mode in a way similar to how it’s done for zoom-out activation.

Let me know your thoughts!

**Comment by draganescu** [comment]
Hi @dhruvang21 sorry for the delay, and thanks for your patience too!

To repro the issues I encountered:

- add a page
- add some patterns
- go to the inspector > Page > Template
- Select template name
- Select show template
- Notice that now we can select the content block
- in write mode the content block has a white toolbar

also

- in a template
- have some template parts
- notice the template parts have a dark toolbar in design mode

I think the problem is we don't target the mode but the "locking" and the issue is to use the dark toolbar _in write mode_ only, eg. when the blocks are locked in content only keep the white toolbar,

This PR also needs a rebase now as the write/design mode have been moved to an experiment. Hope this helps!

**Comment by youknowriad** [comment]
I fixed the issue raised by @oandregal above and addressed some feedback. I think this is ready to go. 

I don't love it personally but I'm ok shipping and trying it.

**Review by youknowriad — APPROVED** [review]
